engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/TypeDefinitionSet.scala

```diff
@@ -0,0 +1,35 @@
+package pl.touk.nussknacker.engine
+
+import org.springframework.expression.spel.SpelNode
+import pl.touk.nussknacker.engine.definition.{DefinitionExtractor, ProcessDefinitionExtractor, TypeInfos}
+import pl.touk.nussknacker.engine.spel.ast.SpelAst.RichSpelNode
+
+
+case class TypeDefinitionSet(typeDefinitions: Set[TypeInfos.ClazzDefinition] = Set()) {
+
+  def validateTypeReference(spelNode: SpelNode): Boolean = {
+
+    val spelNodeChildAST = spelNode.children.headOption.getOrElse(throw new Exception("SpelNode has no children")).toStringAST
+
+    typeDefinitions.exists(typeDefinition => typeDefinition.clazzName.display.equals(spelNodeChildAST))
```

Comment:
`clazzName.display` is not a good choice. Display is only for presentation - not for operational usage. Let's add some method like `javaClassName` to `TypedClass`  that will do `ReflectUtils.fixedClassSimpleNameWithoutParentModule(ClassUtils.primitiveToWrapper(klass))`

Reply:
Added the javaClassName method, thanks

Reply:
also changed TypingResult to TypedClass in ClazzDefinition and implemented the method there